modin/core/storage_formats/base/query_compiler.py

```diff
@@ -4521,6 +4521,59 @@ def has_multiindex(self, axis=0):
         assert axis == 1
         return isinstance(self.columns, pandas.MultiIndex)
 
+    @property
+    def frame_has_materialized_dtypes(self) -> bool:
```

Comment:
@anmyachev, @arunjose696, why didn't we add similar methods for operating on columns?

Reply:
Apparently the relevant code was simply not touched when working on the small query compiler. There is no reason not to do this for columns.

Reply:
I would propose to do this for consistency. @arunjose696, would you take on this?